python/ray/experimental/workflow/api.py

```diff
@@ -56,4 +59,32 @@ def run(entry_workflow: Workflow, workflow_root_dir=None,
     return output
 
 
+# TODO(suquark): support recovery with ObjectRef inputs.
+
+
+def resume(workflow_id: str, workflow_root_dir=None) -> ray.ObjectRef:
+    """
+    Resume a workflow asynchronously. This workflow maybe fail previously.
```

Comment:
Can we raise an error if you try to call run() on an existing workflow? Maybe we can also add a run_or_resume() call.